libs/blocks/merch-cards/merch-cards.js

```diff
@@ -0,0 +1,257 @@
+  const fragment = document.createRange().createContextualFragment(cards);
+  const cardsRoot = fragment.firstElementChild;
+  await makePause();
+  // Replace placeholders
+  cardsRoot.innerHTML = await replaceText(cardsRoot.innerHTML, config);
+  await makePause();
+  const autoBlocks = await decorateLinks(cardsRoot).map(loadBlock);
+  await Promise.all(autoBlocks);
+  await makePause();
+  const blocks = [...cardsRoot.querySelectorAll(':scope > div')].map(loadBlock);
+
+  // process merch card blocks in batches of 3.
+  const batchSize = 3;
+  for (let i = 0; i < blocks.length; i += batchSize) {
+    const batch = blocks.slice(i, i + batchSize);
+    await Promise.all(batch);
+    await makePause();
+  }
+
+  if (filtered) {
+    filterMerchCards(cardsRoot, filtered);
+  }
+  // re-order cards, update card filters
+  [...cardsRoot.children].filter((card) => card.tagName === 'MERCH-CARD').forEach((merchCard) => {
+    const filters = { ...merchCard.filters };
+    Object.keys(filters).forEach((key) => {
+      const preference = preferences[key];
+      if (!preference) return;
+      preference
+        .forEach(([cardTitle, cardSize], index) => {
+          if (merchCard.title === cardTitle) {
+            filters[key] = { order: index, size: cardSize };
+          }
+        });
+    });
+    merchCard.filters = filters;
+  });
+  return cardsRoot;
+}
+
+export default async function main(el) {
+  if (el.classList.length < 2) {
+    return fail(el, 'Missing collection type');
+  }
+
+  const config = getConfig();
+  if (!config.queryIndexCardPath) {
+    return fail(el, 'Missing queryIndexCardPath config');
+  }
+
+  const { miloLibs } = getConfig();
+  loadStyle(`${miloLibs}/blocks/merch/merch.css`);
+  loadStyle(`${miloLibs}/blocks/merch-card/merch-card.css`);
+
+  const attributes = { filter: 'all' };
+  const settingsEl = el.firstElementChild?.firstElementChild;
+
+  const filtered = settingsEl?.firstElementChild?.tagName === 'STRONG';
+
+  if (!filtered) {
+    await Promise.all([
+      import(`${miloLibs}/features/spectrum-web-components/dist/theme.js`),
+      import(`${miloLibs}/features/spectrum-web-components/dist/button.js`),
+      import(`${miloLibs}/features/spectrum-web-components/dist/search.js`),
+      import(`${miloLibs}/features/spectrum-web-components/dist/overlay.js`),
+      import(`${miloLibs}/features/spectrum-web-components/dist/menu.js`),
+      import(`${miloLibs}/features/spectrum-web-components/dist/popover.js`),
+    ]);
+  }
+
+  const preferences = {};
+
+  if (settingsEl) {
+    const [filter, limit = 24] = settingsEl
+      .textContent.split(',').map((s) => s.trim());
+
+    settingsEl.parentElement.remove();
+
+    if (filtered) {
+      attributes.filtered = filter;
+      preferences[filter] = parsePreferences(el.querySelectorAll('p'));
+    } else {
+      if (filter) {
+        attributes.filter = filter;
```

Comment:
find it a bit confusing that both `attributes.filtered` and `attributes.filter` are set to `filter`

`attributes.filtered` sound like a true/false flag but in reality its not?

Reply:
maybe we could have a short readme describing what is held in 'filtered' and what in 'filter'